Source/Modules/app_view_models/Edit/Plugins/Sampler/SynthSamplerViewModel.h

```diff
@@ -6,13 +6,30 @@ const juce::Identifier SYNTH_SAMPLER_VIEW_STATE("SYNTH_SAMPLER_VIEW_STATE");
 
 }
 
-class SynthSamplerViewModel : public app_view_models::SamplerViewModel {
+class SynthSamplerViewModel
+    : public app_view_models::SamplerViewModel,
+      public app_view_models::SynthSamplerState::SynthStateListener {
   public:
     SynthSamplerViewModel(tracktion::SamplerPlugin *sampler);
 
+    void enterDir() override;
+    bool isDir() override;
+
+    juce::String getTitle();
+
     juce::StringArray getItemNames() override;
+    juce::String getSelectedItemName() override;
 
+    void fileChanged(juce::File newFile) override;
     void selectedIndexChanged(int newIndex) override;
+
+  protected:
+    juce::Array<juce::File> files;
+    SynthSamplerState curFileState;
```

Comment:
id just call this state, while it only holds current file stuff for now, that could change in the future.

Reply:
renamed to curState